### DISCUSSION TITLE
Why is this popular

### DISCUSSION THREAD

**Discussion by Cytranics** [AntonOsika/gpt-engineer#351] [General]
Removed I'm an ass.

**Comment by mach7** [comment]
> It's 3-6 API calls max. No reasoning, not Autonomous. It's a single response code creator. Congrats on the viral, but I've been doing the same thing with 5 API calls. Prompts were written by either a non English speaker or a 12 year old.
> 
> 1. Prompts are meh
> 2. Code is meh
> 3. No token managed(not needed you'll never get a full response.
> 4. Zero error handling
> 5. Zero compliance with any standard.
> 6. Zero features.
> 7. No autonomy
> 8. Unneeded file database to load txt files
> 9. Multiple functions to accomplish the same exact thing.
> 
> Meh. I would say it was created by chatgpt but even chatgpt will never make a function called faystem, fuser Good luck, you have the noobs behind you 100%.

I'd be interested to see how you're doing this with the API calls. I haven't been able to get anything close to what I'm wanting the output of this to be.

**Comment by clickbrain** [comment]
Your version of this is so much better! Thanks for sharing your repository... 

I often wonder what it is that would make someone randomly 💩 on the work of others? I mean, what does that get you and why do you think it is that you chose that?

Something a lot of devs fail to understand is that the adoption or popularity of something is not directly related to their perceived quality of the code, but rather the magic the user feels when running it and the outcomes that match their needs. 

It sounds like you have some ideas for making something more magical and more powerful. You had 5 choices when you came here:

1. Join this community and help make it better
2. Fork it and make something better
3. Create something brilliant on your own and make it available to the world
4. Create something you think is better and keep it to yourself (kinda boring and selfish)
5. Throw 💩 at random strangers in Github trying to show how brilliant you

Only one of those choices is a waste of your life. 

Here's a rewrite of your post that provides actual value in the world:

It is really cool how much traction this has gotten. Obviously users are really looking for a solution to write code in a better way. I have some ideas for improving upon it that I am going to post here and then I will submit the code. Thank you for releasing this to the world @AntonOsika so everyone could contribute to make it better. 

1. The prompts here are a good start. I've been using some prompts for coding that have yielded amazing results and have a number of ideas for making these even better. Here are some examples of the prompts, but I'll submit them as well: 
2. I see some specific things in the code that can be improved upon to make it perform better and will submit those. 
3. I'd like to see token management done differently. 
4. After reviewing the code, it seems like we could implement error handling to deal with some of the issues people are encountering. I'll compile a list of the most common and submit for handling. 
6. I am a bit of a stickler for standards and here are some ways we can make the codebase more standards compliant so that we will have less problems later with contributors. I prefer to follow X, but would love to discuss the possibilities. 
7. There are so many new features I can see being added to this. Can't wait to dive in. 
8. Was there consideration given to making this be autonomous and able to work through things more on its own? I think users will love something that takes the code and specification generation a bit further, but keeps in mind user costs. 
9. Have you consider using X for storing data?
10. I noticed that there are multiple functions that we could probably consolidate. Will send suggestions.

**Comment by Cytranics** [comment]
Apologies to the GitHub community and the creator, I've removed the post. You are right, my anger stemmed from Twitter verse calling this the next big thing since sliced bread. Should not have taken it out on the creator.

Sent from Nine<http://www.9folders.com/>
________________________________

**Reply by clickbrain** [reply]
That is awesome man. Thank you for doing that. We all have our crappy moments, so really appreciate that you could respond that way. Apologies if my reply was a bit to snarky.

**Reply by clickbrain** [reply]
That said, it sounded like in the core of. your messaging that you have some great ideas for improving. I hope you'll share some of those, because I know right now the team is pretty overwhelmed because of the hype and somebody like you that has specifically tackled these issues could be very valuable to make gpt-engineer better.

**Reply by crdann** [reply]
LMAO! those god damn AI threaders man, infuriating at times

“bye bye ChatGPT”
“Theres a new kid on the block”
“GPT4 is old news”
